jupyterhub/services/service.py

```diff
@@ -342,12 +344,20 @@ def _default_redirect_uri(self):
             return ''
         return self.host + url_path_join(self.prefix, 'oauth_callback')
 
+    @observe("oauth_client_id", "oauth_redirect_uri")
```

Comment:
There is also the oauth config `oauth_no_confirm`, maybe it belongs here also?

Reply:
Good call, added.